ir/shared/src/main/scala/org/scalajs/ir/Types.scala

```diff
@@ -302,41 +286,30 @@ object Types {
      *  For `NullType` and `NothingType`, the char codes are `'N'` and `'E'`,
      *  respectively.
      */
-    val charCode: Char = tpe match {
-      case VoidType    => 'V'
-      case BooleanType => 'Z'
-      case CharType    => 'C'
-      case ByteType    => 'B'
-      case ShortType   => 'S'
-      case IntType     => 'I'
-      case LongType    => 'J'
-      case FloatType   => 'F'
-      case DoubleType  => 'D'
-      case NullType    => 'N'
-      case NothingType => 'E'
-    }
+    val charCode: Char = charCodeInit
+
+    // Stable hash code, corresponding to reference equality
+    override def hashCode(): Int = charCode.##
```

Comment:
This really scares me. Especially since looking 5 mins at analyzer code did not make it clear to me where we rely on the hash code being stable.

I'm OK with doing this as a stop gap to get this merged (it's not like things are worse after this PR, it's just more explicit that something is broken), but let's file a follow-up issue to figure out what is happening here.